src/admin/dumb_components/sub_due_section.js

```diff
@@ -1,21 +1,59 @@
 import React from 'react'
 
+const { check_tests, date, exists } = require('app/validate')
+import SubDueForm from './sub_due_form.js'
+import { assoc, reduce, unapply, converge, mergeAll, reverse, split, compose, join, map } from 'ramda'
+
+const format_date = compose(join('-'), reverse, split('/'))
+
 export default ({ fetch_sub_due, component, checker, ...props }) => {
-  const send_request = (e) => {
-    e.preventDefault();
-    const [ start, end ] = e.target
-    fetch_sub_due({ start: start.value, end: end.value })
+  const send_request = (data) => {
+    fetch_sub_due(map(format_date, data))
   }
   return (
     <div>
       {checker
         ? component(props)
-        : <form className='date-boundaries-form' onSubmit={send_request}>
-            <input type='text' placeholder='From date' />
-            <input type='text' placeholder='To date' />
-            <button type='submit'>{`Submit Subscription's Due`}</button>
-          </form>
+        : <SubDueForm
+            fields={fields}
+            onSubmit={send_request}
+            validate={validate}
+          />
       }
     </div>
   )
 }
+
+const fields =
+[ 'start'
+, 'end'
+]
+
+const validate = values => {
```

Comment:
Cool. I think it makes sense to carry on with the way we've organised this elsewhere and make a `sub_due_section` file in `form_fields` with all this stuff in